### ISSUE TITLE
[1.11-rc2] docker client is sending application/json type with invalid blank payload

### ISSUE THREAD

**Issue by bboreham** [docker/docker#21725]
See https://github.com/docker/engine-api/pull/186; this breaks compatibility with some programs parsing Docker requests; e.g. https://github.com/weaveworks/weave/issues/2109

**Output of `docker version`:**

```
Client:
 Version:      1.11.0-rc2
 API version:  1.23
 Go version:   go1.5.3
 Git commit:   aa9f9c8-unsupported
 Built:        Fri Apr  1 16:26:27 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0-rc2
 API version:  1.23
 Go version:   go1.5.3
 Git commit:   aa9f9c8-unsupported
 Built:        Fri Apr  1 16:26:27 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 283
Server Version: 1.11.0-rc2
Storage Driver: aufs
 Root Dir: /home/docker/aufs
 Backing Filesystem: extfs
 Dirs: 326
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: null host weavemesh bridge
Kernel Version: 3.13.0-83-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.861 GiB
Name: bbvlx1
ID: IVMU:DIYM:GNIF:DNSW:GCHY:42Z5:R573:MDYB:YWOK:W57O:P4WY:2LQS
Docker Root Dir: /home/docker
Debug mode (client): false
Debug mode (server): false
Username: bboreham
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Steps to reproduce the issue:**
1. Install Weave Net https://github.com/weaveworks/weave
2. `weave launch`
3. `eval $(weave env)`
4. `docker run --net=none -ti alpine /bin/sh`

**Describe the results you received:**

`docker: Error response from daemon: EOF.`

**Describe the results you expected:**

Running shell inside Alpine container.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Comment by icecrime** [comment]
Thanks @bboreham: this looks like a blocker to me, although it's minor enough to make it into RC4 rather than RC3 IMHO.

**Comment by icecrime** [comment]
Closed by #21788 (right @thaJeztah?).

**Comment by thaJeztah** [comment]
Oh! Good catch, yes, forgot to include the "fixes" :blush:
